tb_plugin/torch_tb_profiler/profiler/trace.py

```diff
@@ -33,92 +34,32 @@ def __init__(self, type, data):
         self.duration = data.get("dur", None)
         self.pid = data.get("pid", None)
         self.tid = data.get("tid", None)
-        self.args = data.get("args", None)
-
-    def to_dict(self):
-        return vars(self)
-
-
-class NetEvent(TraceEvent):
-    def __init__(self, data):
-        super(NetEvent, self).__init__(EventTypes.NET, data)
-
-
-class OperatorEvent(TraceEvent):
-    def __init__(self, data):
-        super(OperatorEvent, self).__init__(EventTypes.OPERATOR, data)
+        self.args = data.get("args", {})
 
 
 class ProfilerStepEvent(TraceEvent):
     def __init__(self, data):
-        super(ProfilerStepEvent, self).__init__(EventTypes.PROFILER_STEP, data)
+        super().__init__(EventTypes.PROFILER_STEP, data)
         # torch.profiler.profile.step will invoke record_function with name like "ProfilerStep#5"
         self.step = int(self.name.split("#")[1])
 
 
-class RuntimeEvent(TraceEvent):
-    def __init__(self, data):
-        super(RuntimeEvent, self).__init__(EventTypes.RUNTIME, data)
-
-
-class KernelEvent(TraceEvent):
-    def __init__(self, data):
-        super(KernelEvent, self).__init__(EventTypes.KERNEL, data)
-
-
-class MemcpyEvent(TraceEvent):
-    def __init__(self, data):
-        super(MemcpyEvent, self).__init__(EventTypes.MEMCPY, data)
-
-
-class MemsetEvent(TraceEvent):
-    def __init__(self, data):
-        super(MemsetEvent, self).__init__(EventTypes.MEMSET, data)
-
-
-class PythonEvent(TraceEvent):
-    def __init__(self, data):
-        super(PythonEvent, self).__init__(EventTypes.PYTHON, data)
-
-
-class EventParser(object):
-    def __init__(self):
-        self._handlers = {
-            "X": {
-                "Net": NetEvent,
-                "Operator": self._parse_operator_event,
-                "Runtime": RuntimeEvent,
-                "Kernel": KernelEvent,
-                "Memcpy": MemcpyEvent,
-                "Memset": MemsetEvent,
-                "Python": PythonEvent,
-            }
-        }
-
-    def _get_handler(self, type=None, category=None):
-        handlers = self._handlers.get(type, None)
-        if handlers is None:
+def create_event(event):
+    try:
+        type = event.get("ph")
```

Comment:
type = event.get("ph", None) is unnecessary?

Reply:
From https://docs.python.org/3/library/stdtypes.html#dict.get, event.get("ph") is same with event.get("ph", None)